### PR TITLE
Rebuild for openmpi 5, add wget dep, fix ftp.unidata refs, increase ctest timeout

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

* rebuild openmpi5 and applies updated env vars
* add wget dep needed for testing scripts
* fix now-defunct ftp.unidata.ucar.edu references in testing
* increase default ctest timeout to hopefully help slow mac builds

closes #193

### CODE REVIEW

**Comment by xylar** [comment]
It seems like this got missed in:
https://conda-forge.org/status/migration/openmpi5
Does anyone have time to report this to the bot folks?  I'm a bit too pressed for time to figure out where to report it right now.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge/libnetcdf, anyone able to investigate the failing tests?  They seem unrelated to OpenMPI 5 and rather to be due to missing files or something:
```
 53/246 Test  #47: ncdump_tst_grp_spec ...................   Passed    0.05 sec
test 48
        Start  48: ncdump_tst_nccopy5

48: Test command: /usr/bin/bash "-c" "export srcdir=$SRC_DIR/ncdump;export TOPSRCDIR=$SRC_DIR;$SRC_DIR/build-shared/ncdump/tst_nccopy5.sh tst_nccopy5.sh "
48: Working Directory: $SRC_DIR/build-shared/ncdump
48: Test timeout computed to be: 1500
48: 
48: *** Creating chunkable test file tmp_nc5_base.nc...
48: ok.
48: *** Tests successful!
48: *** Test nccopy -c with per-variable chunking; classic->enhanced
4: Could not ftp AMSR_E_L2_Rain_V10_200905312326_A.hdf.gz
4: $SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 47: return: can only `return' from a function or sourced script
 54/246 Test   #4: hdf4_test_run_get_hdf4_files ..........***Failed   20.84 sec

Getting HDF4 sample files from Unidata FTP site...
Getting HDF4 test files AMSR_E_L2_Rain_V10_200905312326_A.hdf AMSR_E_L3_DailyLand_V06_20020619.hdf     MYD29.A2009152.0000.005.2009153124331.hdf MYD29.A2002185.0000.005.2007160150627.hdf     MOD29.A2000055.0005.005.2006267200024.hdf
$SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 22: wget: command not found
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: ftp.unidata.ucar.edu; Unknown error
wget failed: try 1
$SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 22: wget: command not found
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: ftp.unidata.ucar.edu; Unknown error
wget failed: try 2
$SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 22: wget: command not found
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: ftp.unidata.ucar.edu; Unknown error
wget failed: try 3
$SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 22: wget: command not found
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: ftp.unidata.ucar.edu; Unknown error
wget failed: try 4
Could not ftp AMSR_E_L2_Rain_V10_200905312326_A.hdf.gz
$SRC_DIR/build-shared/hdf4_test/run_get_hdf4_files.sh: line 47: return: can only `return' from a function or sourced script

test 49
        Start  49: ncdump_tst_inttags4

49: Test command: /usr/bin/bash "-c" "export srcdir=$SRC_DIR/ncdump;export TOPSRCDIR=$SRC_DIR;$SRC_DIR/build-shared/ncdump/tst_inttags4.sh tst_inttags4.sh "
49: Working Directory: $SRC_DIR/build-shared/ncdump
49: Test timeout computed to be: 1500
49: *** Test netcdf-4 integer constant suffixes
49: *** creating inttags4.nc from inttags4.cdl...
49: *** creating tst_inttags4.cdl from inttags4.nc...
49: *** comparing tst_inttags4.cdl to ref_inttags4.nc...
 55/246 Test  #49: ncdump_tst_inttags4 ...................   Passed    0.05 sec

```

**Comment by akrherz** [comment]
Appears to be a simple `wget: command not found` , will push a change now

**Comment by akrherz** [comment]
Eh, no perms to push @xylar

**Comment by xylar** [comment]
@akrherz, it doesn't look like you're a maintainer of this feedstock. I can add you as a collaborator on my fork.

**Comment by xylar** [comment]
Oh, I'm really sick this week and it's clear that my head is just too foggy.  `command not found`!  That should have been a great clue!  I was so stuck on the URLs themselves being the issue.

**Comment by xylar** [comment]
Thanks @akrherz!

**Comment by akrherz** [comment]
Now the error are updates Unidata did
```
Could not resolve host: ftp.unidata.ucar.edu; Unknown error
```
They are dangerously close to a new libnetcdf release, so maybe wait a bit more?

**Comment by xylar** [comment]
> They are dangerously close to a new libnetcdf release, so maybe wait a bit more?

Okay, but that'll put the whole OpenMPI migration on hold, right?

**Comment by xylar** [comment]
Thanks @akrherz!  That seems to be doing the trick!

**Comment by akrherz** [comment]
@xylar , I don't see an openmpi migration applied here?  I see this PR is green, but I think this needs updated.  Doing so now.

**Comment by akrherz** [comment]
@conda-forge-admin please rerender

**Comment by xylar** [comment]
@akrherz, good catch!  I added the migrator file locally but forgot to `git add` it.

**Comment by xylar** [comment]
Okay, I see.  We need to migrate `libpnetcdf` first.  Also missed by the bot???

**Comment by akrherz** [comment]
yes, what @xylar said.
```
The following packages are incompatible
├─ hdf5 *,1.14.3.* mpi_openmpi_* is installable with the potential options
│  ├─ hdf5 1.14.3 would require
│  │  └─ openmpi >=4.1.6,<5.0a0 , which can be installed;
│  └─ hdf5 1.14.3 would require
│     └─ openmpi >=5.0.3,<6.0a0 , which requires
│        └─ mpi 1.0 openmpi, which can be installed;
├─ libpnetcdf is installable with the potential options
│  ├─ libpnetcdf 1.12.3 would require
│  │  └─ openmpi >=4.1.5,<5.0a0 , which can be installed;
│  ├─ libpnetcdf [1.12.3|1.13.0] would require
│  │  └─ mpich [>=4.1.1,<5.0a0 |>=4.2.0,<5.0a0 ], which requires
│  │     └─ mpi 1.0 mpich, which conflicts with any installable versions previously reported;
│  └─ libpnetcdf 1.13.0 would require
│     └─ openmpi >=4.1.6,<5.0a0 , which can be installed;
└─ openmpi 5.* , which cannot be installed (as previously explained).
```

**Comment by akrherz** [comment]
conda-forge/libpnetcdf-feedstock#7 is merged, so awaiting CDN to restart this :)

**Comment by akrherz** [comment]
@conda-forge-admin please restart ci

and oh my, this [conda-metadata-app](https://conda-metadata-app.streamlit.app/) is nice.

**Comment by akrherz** [comment]
oye, transient failure or related to open-mpi/ompi#12431
```
prte_plm_base_select failed
    --> Returned value  (-46) instead of PRTE_SUCCESS
```

**Comment by akrherz** [comment]
@conda-forge-admin please rerender

**Comment by akrherz** [comment]
If this works, I will sort of be shocked ... but now I stumbled into conda-forge/openmpi-feedstock#152

If this goes green, will instead try the `export OMPI_MCA_plm_ssh_agent=false` approach

**Comment by xylar** [comment]
🤞🏻

**Comment by xylar** [comment]
I've seen the openssh thing be necessary on other feedstocks.  I assumed it was always required, so was surprised it wasn't here, too.

**Comment by xylar** [comment]
https://github.com/conda-forge/netcdf4-feedstock/blob/69b6fd7f657124ecf78064422f6013c0ace5e16a/recipe/meta.yaml#L76
https://github.com/conda-forge/netcdf-cxx4-feedstock/blob/74022cfdfbefec51cdcf1c9f78c9c5d49b1dcae9/recipe/meta.yaml#L70
https://github.com/conda-forge/esmf-feedstock/blob/d457f25a89d2d143c60fc32889496a473c36226b/recipe/meta.yaml#L94

We seem to typically add it to run, rather than build I see.

**Comment by akrherz** [comment]
> We seem to typically add it to run, rather than build I see.

Fair.  Am going to hope to avoid it completely, I have the next commit teed up, awaiting a green mpiopenmpi job first :)

**Comment by xylar** [comment]
No dice :-(

**Comment by akrherz** [comment]
> No dice :-(

If this doesn't work, will try your `openssh` in `run` dep.

**Comment by akrherz** [comment]
This doesn't make sense to me as the testing is done within the build phase, so if this somehow works....

**Comment by akrherz** [comment]
> Sorry for the confusion.

At least putting it in `run` confirmed what I assumed should be happening, lets try `host`, for funzies.

**Comment by akrherz** [comment]
Holy cow, I need to review this morning and back out some debugging before somebody thinks about merging...  @xylar feel free to do that, or I will get to it in a couple of hours

**Comment by akrherz** [comment]
my debugging triggered an unrelated failure
```
 which: no ssh in ($BUILD_PREFIX/bin:$PREFIX/bin:/opt/conda/condabin:$BUILD_PREFIX/bin:$PREFIX/bin:/opt/conda/bin:/opt/conda/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/home/conda/bin)
```
now I think openssh needs to be in `build` , unwinded some debugging :eyes:

**Comment by akrherz** [comment]
If this gets green, I then what to check if we can remove openssh dep and just use `[OMPI_MCA_plm_ssh_agent=false](https://github.com/conda-forge/libnetcdf-feedstock/pull/192/commits/ba7ca305067c195a0a37ee1761c0a50d9419a8db)`.  It continues to be very confusing with issues and conflating `plm` and `pml`.  I'll have much putzing done in a few hours and can then move this forward, hopefully.

**Comment by akrherz** [comment]
osx_64 failure may be transient timeout

> Test #174: nc_test4_run_par_test .................***Timeout 1500.22 sec

File: recipe/build.sh [hunk 1]
```diff
@@ -11,7 +11,8 @@ if [[ ! -z "$mpi" && "$mpi" != "nompi" ]]; then
   export TESTPROC=4
   export OMPI_MCA_rmaps_base_oversubscribe=yes
   export OMPI_MCA_btl=self,tcp
-  export OMPI_MCA_plm=isolated
+  # for openmpi5 within containers without SSH
+  export OMPI_MCA_plm_ssh_agent=false
   export OMPI_MCA_rmaps_base_oversubscribe=yes
   export OMPI_MCA_btl_vader_single_copy_mechanism=none
```
Comment (minrk) [inline recipe/build.sh:17]: openmpi 5 changed the names and values ~all of these variables. Fortunately, the openmpi 5 conda-forge package now sets the right ones needed for conda-build, so you can delete all of these OMPI_MCA variables and it should work.

**Comment by xylar** [comment]
@akrherz, I think it's better to stick with this PR than move to #193.  It just makes things more confusing as far as I'm concerned.  I have brought most of those commits over her and applied @minrk's commit after them.  But with 2 PRs open, it's not clear where folks should contribute.  I'll update the name of this one to match yours.

File: recipe/build.sh (hunk 1 above)
Comment (xylar) [reply recipe/build.sh]: Thanks @minrk!

**Comment by xylar** [comment]
Assuming all the tests pass, I'll merge this later this afternoon (European time).

**Comment by xylar** [comment]
Okay, let's keep this moving along.  Thanks so much @akrherz for working so hard on this and @minrk for your contribution.

**Pull request merged**